sky/packages/sky/lib/widgets/render_toggleable.dart

```diff
@@ -0,0 +1,73 @@
+// Copyright 2015 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:sky' as sky;
+
+//import 'package:sky/painting/shadows.dart';
+//import 'package:sky/theme/shadows.dart';
+import 'package:sky/animation/animated_value.dart';
+import 'package:sky/animation/animation_performance.dart';
+import 'package:sky/animation/curves.dart';
+import 'package:sky/rendering/box.dart';
+import 'package:sky/rendering/object.dart';
+import 'package:sky/widgets/basic.dart';
+import 'package:sky/widgets/widget.dart';
+
+export 'package:sky/widgets/basic.dart' show ValueChanged;
+
+const Duration _kToggleDuration = const Duration(milliseconds: 200);
+
+// RenderToggleable is a base class for material style toggleable controls with
+// toggle animations. It handles storing the current value, dispatching
+// ValueChanged on a tap gesture and driving a changed animation. Subclasses are
+// responsible for painting.
+abstract class RenderToggleable extends RenderConstrainedBox {
```

Comment:
I wonder if we should put this in the sky/rendering directory.  If we can remove the imports of sky/widgets/*, then the dependencies will work out.  Also, then it might be a sensible place to declare ValueChanged.